lib/Standalone/MapConv2DNhwcHwcfToMatmul.cpp

```diff
@@ -109,24 +109,53 @@ static bool checkMappingToMatmul(linalg::LinalgOp linalgOp) {
   return match;
 }
 
-static bool preConditionsMapConvToGemm(linalg::LinalgOp linalgOp) {
+// Check if the operation looks like a convolution.
+static bool isConvLike(linalg::LinalgOp linalgOp) {
```

Comment:
For example, this here could return `rewriter.notifyMatchFailure()` directly with a more fine-grained error message?

Reply:
Yeah, unclear to me if we can connect this notifyMatchFailure with the transform diagnostic. The error you see comes from the transform not the rewriter.

Reply:
Right, then perhaps a comment for now, per test, as to why things don't work would be fine